src/raft.rs

```diff
@@ -861,6 +869,7 @@ impl<T: Storage> Raft<T> {
 
         let last_index = self.raft_log.last_index();
         let committed = self.raft_log.committed;
+        self.uncommitted_size = 0;
```

Comment:
After a new leader elected, we should also count the uncommitted log of the previous term or `reduce_uncommitted_size` may panic.

Reply:
Done, also add a test for this situation, PTAL:)

Reply:
it seems that [etcd-io/etcd](https://github.com/etcd-io/etcd/blob/master/raft/raft.go#L632) has the same issue? If we don't set this value correctly, raft can not estimate uncommitted size percisely.

eg. 
node1 has 20 bytes of uncommitted data and has 30 for its limit.
But when it becomes leader,  node1's uncommitted size is set to 0.
We can now add 30 more bytes to its log where it should only be 10.

Reply:
Yes, etcd have a [workaround](https://github.com/etcd-io/etcd/blob/cc2b4cd05e6b677c9c73d9ccb7333ebf1e1be7ee/raft/raft.go#L1666-L1670) for this issue.